app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -575,40 +576,89 @@ class HomeActivityTest {
     // Only new users will have no progress for any topics.
-    profileTestHelper.loginToNewUser()
+    profileTestHelper.logIntoNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 2)
+
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        hasGridItemCount(spanCount = 1, position = 3)
+      )
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testHomeActivity_noTopicsStarted_mobileLandspaceDisplaysTopicsIn3Columns() {
+    // Only new users will have no progress for any topics.
+    profileTestHelper.logIntoNewUser()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 3)
+
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        hasGridItemCount(spanCount = 1, position = 3)
+      )
+    }
+  }
+
+  @Config(qualifiers = "sw600dp-port")
+  @Test
+  fun testHomeActivity_noTopicsStarted_tabletPortraitDisplaysTopicsIn3Columns() {
+    // Only new users will have no progress for any topics.
+    profileTestHelper.logIntoNewUser()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 3)
+
       scrollToPosition(position = 3)
       onView(withId(R.id.home_recycler_view)).check(
-        // The "All Topics" section currently should display all four test topics in two rows.
-        hasGridColumnCount(expectedColumnCount = 2)
+        hasGridItemCount(spanCount = 1, position = 3)
       )
     }
   }
 
+  @Config(qualifiers = "sw600dp-land")
   @Test
-  fun testHomeActivity_noTopicsStarted_displaysAllTopicCardsInLandscape() {
+  fun testHomeActivity_noTopicsStarted_tabletLandscapeDisplaysTopicsIn4Columns() {
     // Only new users will have no progress for any topics.
-    profileTestHelper.loginToNewUser()
+    profileTestHelper.logIntoNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
       onView(isRoot()).perform(orientationLandscape())
+      hasGridColumnCount(expectedColumnCount = 4)
+
       scrollToPosition(position = 3)
-      var rowsUsed = 2
-      if (context.resources.getBoolean(R.bool.isTablet)) {
-        // A landscape tablet can display the four test topics in one row.
-        rowsUsed = 1
-      }
       onView(withId(R.id.home_recycler_view)).check(
-        // The "All Topics" section currently should display all four test topics.
-        hasGridColumnCount(expectedColumnCount = rowsUsed)
+        hasGridItemCount(spanCount = 1, position = 3)
       )
     }
   }
 
+  @Test
+  fun testHomeActivity_onScrollDown_promotedStoryListViewStillShows() {
```

Comment:
Can you confirm that this fails on both Robolectric & Espresso without the fix in place?

Reply:
You mentioned above that you confirmed this--thanks!